add-ons/istio/istio.addon

```diff
@@ -0,0 +1,35 @@
+# Name: istio
+# Description: Deploys Latest Istio
+# Url: https://github.com/istio/istio
```

Comment:
@kameshsampath you need to add minimum openshift version to use this addon looks like the issue (https://github.com/minishift/minishift-addons/pull/55#issuecomment-352408870) which I saw because I was using 3.6.1 \ which is by default as of now for minishift.

- docs https://docs.openshift.org/latest/minishift/using/addons.html#addon-openshift-version-semantics

Reply:
@praveenkumar @kameshsampath I am still getting same error with `3.7.0` and k8s version `1.7.6`

```
$ minishift addon apply istio --addon-env ISTIO_HOME=$ISTIO_HOME
-- Applying addon 'istio':.....unable to recognize "/home/budhram/istio-0.3.0/install/kubernetes/istio.yaml": no matches for config.istio.io/, Kind=logentry
unable to recognize "/home/budhram/istio-0.3.0/install/kubernetes/istio.yaml": no matches for config.istio.io/, Kind=metric
unable to recognize "/home/budhram/istio-0.3.0/install/kubernetes/istio.yaml": no matches for config.istio.io/, Kind=metric
unable to recognize "/home/budhram/istio-0.3.0/install/kubernetes/istio.yaml": no matches for config.istio.io/, Kind=metric
unable to recognize "/home/budhram/istio-0.3.0/install/kubernetes/istio.yaml": no matches for config.istio.io/, Kind=metric
unable to recognize "/home/budhram/istio-0.3.0/install/kubernetes/istio.yaml": no matches for config.istio.io/, Kind=metric
unable to recognize "/home/budhram/istio-0.3.0/install/kubernetes/istio.yaml": no matches for config.istio.io/, Kind=metric
Error applying the add-on: Error executing command oc create -f #{ISTIO_HOME}/install/kubernetes/istio.yaml.

$ minishift openshift version
openshift v3.7.0+7ed6862
kubernetes v1.7.6+a08f5eeb62
etcd 3.2.8

```

Reply:
@budhrg did you have any previous (failed/success) ? can you please try with new fresh clean install of OCP v3.7 and also using latest client `minishift oc-env` ?

Reply:
Yes @kameshsampath , it was clean build.

I started as
```
$ minishift delete -f
$ rm -rf ~/.minishift

# Start with Istio instructions
```

```
$ minishift oc-env
export PATH="/home/budhram/.minishift/cache/oc/v3.7.0/linux:$PATH"
# Run this command to configure your shell:
# eval $(minishift oc-env)
```

Reply:
@budhrg After you start minishift, wait for some time around 5 mins so that each and every openshift component are up and then try to install/apply and then you will not see that.